### PR TITLE
refactor: dt-31 Modal transition refactor to use Vue transition compatible CSS classes

### LINKED ISSUE #159: Modal transition: refactor to use Vue-transition-compatible CSS classes instead of aria-hidden
**Summary**
The current modal styling is controlled by the setting of the `aria-hidden` attr.  This means that the modal is always rendered in the DOM, even if it's never triggered and made visible.

If the transition was instead accomplished with a set of css classes for the various transition-stages (e.g. animate-in, animate-out), we could map those directly to Vue's transition classes: https://v3.vuejs.org/guide/transitions-enterleave.html#transition-classes.

**Details**
Per discussion w/ @dbecherdialpad on the Handset Modal PR https://github.com/dialpad/firespotter/pull/7210

> I am more convinced that this HsModal component should use LazyShow as its root node so that we don't render the modal content until it's open. It's fine for now but you can imagine a scenario where we have a lot of modals on a page (maybe even one per row in a table!) and it will be pretty inefficient to render them all into the DOM at once instead of when they open.

> The only real change that would be required to do this would be to use a CSS class for the transition instead of the aria-hidden prop.

And from earlier discussion w/ Ted/David in slack:

Something like:
```
.d-modal--animate { sets the transition properties... }
.d-modal--animate-in { transitions to visible }
.d-modal--animate-out { transitions to hidden }
```

Where the various vue transition classes could then be bound to those in the modal component, e.g. `fade-enter-active: d-modal--animate-in`.

### PR DESCRIPTION
**Pull request by josedialpad**
## Description
Issue: https://switchcomm.atlassian.net/browse/DT-31
This PR adds modal transition classes to apply during transition instead of relying on `aria-hidden` prop to transition.

Updated documentation modal examples in order to use these classes.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [ ] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Provide a description what is being changed, extended, or removed.
 - [x] Link to any issue(s) this request is resolving.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Drew Chandler, Joshua Hynes, or Ted Goas.

## Obligatory GIF
CleanShot 2021-08-20 at 10 59 51

### CODE REVIEW

**Comment by dialpad-drew** [comment]
How does this impact the DT6 implementation? Anything we need to note there?

Would we need to add these new classes to the class table?

**Comment by josedialpad** [comment]
@dialpad-drew since with this PR the transition is made with CSS classes, in the DT6 implementation we need to apply the classes during the transition. In the [following PR](https://github.com/dialpad/dialtone-vue/pull/137) the Vue transition wrapper was used with these new transition classes.

> Would we need to add these new classes to the class table?

Yes, and we could add some documentation about it in the dialtone documentation.
Update: added transition classes in Modal classes documentation table https://github.com/dialpad/dialtone/commit/e052c1ca3a72496a18ba5cae47ff7adcad6fc4d3

**Review by dialpad-drew — APPROVED** [review]

**Comment by dialpad-drew** [comment]
Are we waiting for David's approval?

**Comment by josedialpad** [comment]
@dialpad-drew ideally yes. I'd like his review especially in the implementation PR https://github.com/dialpad/dialtone-vue/pull/137 but If we're good with these changes we can merge this and just wait for the review on the implementation PR (probably after Dialtone 6 migration since it's the priority).

File: lib/build/less/components/modal.less [hunk 1]
```diff
@@ -49,11 +49,6 @@
...
     backface-visibility: hidden;
 
-    transition:
-        opacity 100ms var(--ttf-in-out) 10ms,
-        z-index 0s 200ms,
-        visibility 0s 200ms,
-        transform 100ms var(--ttf-in-out) 10ms;
     will-change: visibility, z-index, opacity;
 
     @media (prefers-reduced-motion) {
```
Comment (dbecherdialpad) [inline lib/build/less/components/modal.less]: I think this `prefers-reduced-motion` query needs to be wrapped around the new transition classes below (and make sure the classes work with `prefers-reduced-motion` enabled)

**Review by dbecherdialpad — CHANGES_REQUESTED** [review]

File: lib/build/less/components/modal.less (hunk 1 above)
Comment (josedialpad) [reply lib/build/less/components/modal.less]: Thanks, you're right. Updated to put `prefers-reduced-motion` in the transition classes. https://github.com/dialpad/dialtone/pull/407/commits/17ff5c43ac3dcc33de9998f30820e0bd4d7525b1

> and make sure the classes work with prefers-reduced-motion enabled

The classes work but not the transition as expected.

**Review by dbecherdialpad — APPROVED** [review]

**Pull request merged**